src/components/MeetupGallery.astro

```diff
@@ -0,0 +1,193 @@
+                  loading="lazy"
+                />
+              </div>
+            ))
+        }
+      </div>
+  </div>
+
+  <script define:vars={{ props }}>
+
+    const imageArray = getSplitArray(props.currentAlbum, 3);
+    let counters = [0, 0, 0];
+    let isMouseOver = false;
+
+
+    startAnimation();
+    addEventListenerOnImages();
+
+    function startAnimation(){
+      setInterval(() => {
+        if(!isMouseOver){
+          changeImages();
+        }
+      }, 7000);
+    }
+
+    function changeImages(){
+      changeImgClassName();
+      for(let i=0;i<3;i++){
+        // console.log("loop: " + i + " counter: " + counters[i]);
+       changeImgSrc(i+1, imageArray[i], counters[i]);
+       counters[i]++;
+        if(counters[i] >= imageArray[i].length){
+          counters[i] = 0;
+        }
+      }
+    }
+
+    function addEventListenerOnImages(){
+      const imageContainers = document.querySelectorAll(".photo-container");
+      imageContainers.forEach((imageContainer) => {
+        imageContainer.addEventListener("mouseover", () => {
+          isMouseOver = true;
+        });
+        imageContainer.addEventListener("mouseout", () => {
+          isMouseOver = false;
+        });
+      });
+    }
+
+    function changeImgClassName(){
+      // console.log('swapping classes...');
+      const img1Selector = document.querySelector(".photo-1");
+      const img2Selector = document.querySelector(".photo-2");
+      const img3Selector = document.querySelector(".photo-3");
+
+      if(img1Selector){
+        img1Selector.classList.remove("photo-1");
+        img1Selector.classList.add("photo-2");
+      }
+
+      if(img2Selector){
+        img2Selector.classList.remove("photo-2");
+        img2Selector.classList.add("photo-3");
+      }
+
+      if(img3Selector){
+        img3Selector.classList.remove("photo-3");
+        img3Selector.classList.add("photo-1");
+      }
+    }
+
+    
+    
+
+    function changeImgSrc(imgSelector, array, counter){
+      const img = document.querySelector(".photo-container:nth-child("+imgSelector+") img");
+      
+      if(img){
+        // console.log(`${props.source}/${array[counter]}`);
+        img.style.opacity = "0.1";
+        img.src = `${props.source}/${array[counter]}`;
+        // img.style.opacity = "1";
+        setTimeout(()=>{img.style.opacity = "1"}, 300);
+      }
+    }
+
+    function getSplitArray(array, size) {
+      const splitSize = Math.floor(array.length / size);
+      const resultArray = [];
+      for(let i = 0; i < size; i++) {
+        let arr = [];
+        arr = array.splice(0, splitSize);
+        resultArray.push(arr);
+      }
+      return resultArray;
+    }
+  </script>
+
+  <style lang="scss">
+
+    .main-container{
+        display: grid;
+        place-items: center;
+
+        .container{
+            width: 100px;
+            height: 100px;
+            position: relative;
+
+            .photo-container{
+                width: 231px;
```

Comment:
use a css variable for the size of the container
`--photoSize: 231px`
width: var(--photoSize: 231px);
This lets you do the responsive easily and with less code.